WordPress/gutenberg
# [RN Mobile] - Add helper to wait for screen to load before continuing test

## What?
This PR attempts to fix the flakiness we're seeing on the Android test by waiting for the expected screen to load before continuing the test

Multiple test runs were done on this [draft PR](https://github.com/WordPress/gutenberg/pull/39334) and the results look to be promising

## Why?
To reduce flakiness on Android builds tests. This PR is a follow up to the issue in [this PR](https://github.com/WordPress/gutenberg/pull/39315) 

## How?
This PR removes the implicit waits that are being called during initialization and replaces them with a new `waitForVisible()` function that waits for the screen to load. The wait time is currently set to 25 seconds (adjustable if we decided this is not enough in the future). If the screen loads earlier, it will stop waiting and continue with the test. If the screen does not load within 25 seconds, an error will be thrown. 

Currently being used on editor screen initialization and on `getTextViewForHtmlViewContent()` as we are seeing failures like [this](https://github.com/WordPress/gutenberg/runs/5496510612?check_suite_focus=true) and [this](https://github.com/WordPress/gutenberg/actions/runs/1966940997) that seems to be happening intermittently on that particular function.

## Testing Instructions

1. Run the tests in `React Native E2E Tests (Android)` and `React Native E2E Tests (iOS)` and ensure that tests are passing
